compute/sqlserver/powershell/create-availability-group.ps1

```diff
@@ -0,0 +1,409 @@
+    $objDB = New-Object `
+      -TypeName Microsoft.SqlServer.Management.Smo.Database($objServer, $db_name)
+
+    # Create the primary file group and add it to the database
+    $objPrimaryFG = New-Object `
+      -TypeName Microsoft.SqlServer.Management.Smo.Filegroup($objDB, 'PRIMARY')
+    $objDB.Filegroups.Add($objPrimaryFG)
+
+    # Create a single data file and add it to the Primary filegroup
+    $dataFileName = $db_name + '_Data'
+    $objData = New-Object `
+      -TypeName Microsoft.SqlServer.Management.Smo.DataFile($objPrimaryFG, $dataFileName)
+    $objData.FileName = $sql_data + '\' + $dataFileName + '.mdf'
+    $objData.Size = ($data_size * 1024)
+    $objData.GrowthType = 'KB'
+    $objData.Growth = ($data_growth * 1024)
+    $objData.IsPrimaryFile = 'true'
+    $objPrimaryFG.Files.Add($objData)
+
+    # Create the log file and add it to the database
+    $logName = $db_name + '_Log'
+    $objLog = New-Object Microsoft.SqlServer.Management.Smo.LogFile($objDB, $logName)
+    $objLog.FileName = $sql_log + '\' + $logName + '.ldf'
+    $objLog.Size = ($log_size * 1024)
+    $objLog.GrowthType = 'KB'
+    $objLog.Growth = ($log_growth * 1024)
+    $objDB.LogFiles.Add($objLog)
+
+    # Create the database
+    $objDB.Script()  # Show a script with the command we are about to run
+    $objDB.Create()  # Create the database
+    $objDB.SetOwner('sa')  # Change the owner to sa
+  }
+  else {
+      Write-Host "$(Get-Date) $hostname - Skipping the creation of the $db_name DB"
+  }
+} -ArgumentList $db_name, $sql_data, $sql_log, $log_size, $log_growth, `
+  $data_size, $data_growth
+
+
+
+################################################################################
+# Create a shared folder for initial backup used by Availability Groups
+# We choose Node 2, but it can be any network share accesible by both nodes
+################################################################################
+Invoke-Command -ComputerName $node2_fqdn -Credential $cred -ScriptBlock { 
+  param($sql_backup, $share_name, $domain, $svc_acct, $node1)
+
+  $hostname = [System.Net.Dns]::GetHostName()
+  
+  # Create folder for the backup in Node2.
+  # For this demo script we just use one of the nodes for simplicity
+  Write-Host "$(Get-Date) $hostname - Creating folders: $sql_backup"
+  if (!(Test-Path -Path $sql_backup )) { 
+    New-item -ItemType Directory $sql_backup | Out-Null }
+
+  # Create a Windows share for the folder
+  Write-Host "$(Get-Date) $hostname - Creating Windows Share: $share_name"
+  Get-SmbShare | Where-Object -Property Name -eq $share_name | 
+    Remove-SmbShare -Force
+  New-SMBShare –Name $share_name –Path $sql_backup –FullAccess "$($node1)`$","NT SERVICE\MSSQLSERVER" | Out-Null
+} -ArgumentList $sql_backup, $share_name, $domain, $svc_acct, $node1
+
+
+# Enable CredSSP in local computer
+Write-Host "$(Get-Date) Enable CredSSP Client in local machine"
+Enable-WSManCredSSP Client -DelegateComputer * -Force | Out-Null
+Start-Sleep 15

```

Comment:
Ok.  Sounds like a reasonable solution.  Please say something about the random error in a comment.